File: backend/src/main/java/dev/coldhands/pair/stairs/backend/infrastructure/persistance/entity/CombinationEventEntity.java
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "combination_events")
public class CombinationEventEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "date")
    private LocalDate date;

    @ManyToOne
    @JoinColumn(name = "combination_id")
    private CombinationEntity combination;

    protected CombinationEventEntity() {
    }

    public CombinationEventEntity(LocalDate date, CombinationEntity combination) {
        this.date = date;
        this.combination = combination;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public CombinationEntity getCombination() {
        return combination;
    }

    public void setCombination(CombinationEntity combination) {
        this.combination = combination;
    }
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/infrastructure/persistance/entity/CombinationEventEntity.kt
package dev.coldhands.pair.stairs.backend.infrastructure.persistance.entity

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "combination_events")
class CombinationEventEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "date")
    val date: LocalDate,

    @ManyToOne
    @JoinColumn(name = "combination_id")
    val combination: CombinationEntity
) {
    // todo remove me once everything is kotlin
    constructor(date: LocalDate, combination: CombinationEntity) : this(id = null, date = date, combination = combination)
}